pkg/plugins/backendplugin/manager.go

```diff
@@ -221,32 +225,46 @@ func (m *manager) CallResource(config PluginConfig, c *models.ReqContext, path s
 		URL:     clonedReq.URL.String(),
 		Headers: clonedReq.Header,
 		Body:    body,
-		User:    c.SignedInUser,
+		User:    reqCtx.SignedInUser,
 	}
 
-	stream, err := p.callResource(clonedReq.Context(), req)
+	err = InstrumentPluginRequest(p.id, "resource", func() error {
+		stream, err := p.callResource(clonedReq.Context(), req)
+		if err != nil {
+			return errutil.Wrap("Failed to call resource", err)
+		}
+
+		err = flushStream(p, stream, reqCtx)
+		if err == ErrStreamDrained {
+			return nil
+		}
+
+		return err
+	})
+
 	if err != nil {
-		c.JsonApiErr(500, "Failed to call resource", err)
-		return
+		reqCtx.JsonApiErr(500, "Failed to ", err)
 	}
+}
 
+func flushStream(plugin *BackendPlugin, stream callResourceResultStream, reqCtx *models.ReqContext) error {
 	processedStreams := 0
 
 	for {
 		resp, err := stream.Recv()
 		if err == io.EOF {
 			if processedStreams == 0 {
-				c.JsonApiErr(500, "Received empty resource response ", nil)
+				return errors.New("Received empty resource response")
 			}
-			return
+			return ErrStreamDrained
```

Comment:
Nit, not sure this custom error is needed any longer. Wouldn't return nil instead be more appropriate?

Reply:
good catch!